doc/architecture/decisions/0004-integration-patterns.md

```diff
@@ -0,0 +1,237 @@
+# Integration Patterns
+
+- Status: Draft
+- Date: December 2022
+
+## Context
+
+PlanX allows service designers to create, edit and publish digital services to the public. Where these services have a back-office processing stage, session data must be made available to the appropriate **downstream consumers** (3rd party platforms) after the completion of each **session** (a user's journey through a service). An **integration** is any pattern that allows PlanX to securely pass session data to these downstream consumers when configured to do so by the "Send" component within a service.
+
+The PlanX platform is currently scaling the number of teams using the platform, some of whom will require integrations to downstream consumers which are not currently supported. Open Digital Planning has also announced a fund for integrations with PlanX/BOPs/Digital Land. Both of these factors mean that a clear and robust set of integration patterns is required in the near future.
+
```

Comment:
The definition of **integration** here is an interesting one for me: in this document we're (rightly) focusing on essentially one-directional _submission_ integrations, but I think existing partners (eg DLUHC) also want to say that "Planx integrates with Digital Land" (which is currently an integration in the other direction, but perhaps in the future is bi-directional - eg we feed back info to digital land if an applicant contests a planning constraint). 

I'm wondering if we can be a more explicit about some of this context up front here, and then shift into the subset of integrations that support the submission of Planx applications. I'll think more on ways to write that.